protocol/xmrmaker/claim.go

```diff
@@ -147,19 +162,55 @@ func (s *swapState) discoverRelayersAndClaim() (ethcommon.Hash, error) {
 			continue
 		}
 
-		return resp.TxHash, nil
+		log.Infof("DHT relayer's claim validated block=%d gas-used=%d tx=%s",
+			receipt.BlockNumber, receipt.GasUsed, receipt.TxHash)
+
+		return receipt, nil
 	}
 
-	return ethcommon.Hash{}, errors.New("failed to submit transaction to any relayer")
+	return nil, errors.New("failed to relay claim with any non-counterparty relayer")
+}
+
+// claimWithRelay first attempts to relay with the help of the XMR taker and, if
+// that fails, tries all relayers advertising in the DHT. Note that the receipt
+// returned is for a transaction created by the remote relayer, not by us.
+func (s *swapState) claimWithRelay() (*ethtypes.Receipt, error) {
+	forwarderAddress, err := s.Contract().TrustedForwarder(&bind.CallOpts{Context: s.ctx})
+	if err != nil {
+		return nil, err
+	}
+
+	secret := s.getSecret()
+
+	request, err := relayer.CreateRelayClaimRequest(
+		s.ctx,
+		s.ETHClient().PrivateKey(),
+		s.ETHClient().Raw(),
+		s.contractAddr,
+		forwarderAddress,
+		s.contractSwap,
+		&secret,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	receipt, err := s.relayClaimWithXMRTaker(request)
+	if err != nil {
+		log.Warnf("failed to relay claim with XMR taker: %s", err)
```

Comment:
```suggestion
		log.Warnf("failed to relay claim with counterparty: %w", err)
```

Reply:
The code shifted around when I made the Taker the relayer of last resort. Since we know which counterparty was relaying, isn't the more specific message better? It matches the function names and other logged messages with the more specific wording.

Reply:
Resolving. This log ended up in a new location and I used the suggested wording on that location.